pkg/controllers/dlb/controller.go

```diff
@@ -93,7 +94,7 @@ func (c *controller) UpdateDaemonSet(rawObj client.Object, ds *apps.DaemonSet) (
 	}
 
 	if !reflect.DeepEqual(ds.Spec.Template.Spec.NodeSelector, dp.Spec.NodeSelector) {
```

Comment:
btw, we don't need to call DeepEqual anymore. Here I'd suggest to use the same code as above:
```
if len(dp.Spec.NodeSelector) > 0 {
        ds.Spec.Template.Spec.NodeSelector = dp.Spec.NodeSelector
}
```

Reply:
Actually it's a bit more compicated. as @mythi suggested we should also restore default value if dp.Spec.NodeSelector is empty.

Reply:
I did it, but then what about "updated = true"?
It's always true after the function is run.

Reply:
@bart0sh I put just
else {....NodeSelector = default from yaml file}
Isn't it what he meant for updatedaemonset function?

Reply:
yes, that's what he suggested and I pointed out to.

Reply:
> I did it, but then what about "updated = true"?
> It's always true after the function is run.

Good point. That means we still need to compare them with DeepEqual